application/F3DOptionsParser.cxx

```diff
@@ -279,7 +260,67 @@ void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& o
 {
   inputs.clear(); /* needed because this function is called multiple times */
 
-  this->FilePathForConfigBlock = std::move(filePathForConfigBlock);
+  /* start with an empty config ... */
+  ConfigDict tmpConfig;
+
+  const auto update = [&](const ConfigDiff& config)
+  {
+    for (const auto& [key, maybeValue] : config)
+    {
+      const std::string del = tmpConfig.count(key) ? key + ": " + tmpConfig[key]
+        : !maybeValue.has_value()                  ? key
+                                                   : "";
+      const std::string add = maybeValue.has_value() ? key + ": " + maybeValue.value() : "";
+      if (add == del && !add.empty())
+      {
+        f3d::log::debug("= ", add);
+      }
+      else
+      {
+        if (!del.empty())
+        {
+          f3d::log::debug("- ", del);
+        }
+        if (!add.empty())
+        {
+          f3d::log::debug("+ ", add);
+        }
+      }
+
+      if (maybeValue.has_value())
+      {
+        tmpConfig[key] = maybeValue.value();
+      }
+      else
+      {
+        tmpConfig.erase(key);
+      }
+    }
+  };
+
+  /* apply global entries */
+  for (const auto& [conf, source, pattern] : this->GlobalConfigEntries)
+  {
+    f3d::log::debug("using `", pattern, "` config from ", source.string());
+    update(conf);
+  }
+
+  /* ... then go through matching regex configs and override */
+  if (!filePathForConfigBlock.empty())
+  {
+    for (auto const& [conf, source, pattern] : this->RegexConfigEntries)
+    {
+      std::regex re(pattern, std::regex_constants::icase);
+      std::smatch matches;
+      if (std::regex_match(filePathForConfigBlock, matches, re))
+      {
+        f3d::log::debug("using `", pattern, "` config from ", source.string());
+        update(conf);
+      }
+    }
+
+    this->CurrentFileConfig = tmpConfig;
```

Comment:
what does this contains ?

Reply:
it contains the final configuration key/values (merged from global blocks and matched regex blocks) so that `GetOptionConfig()` can lookup directly into that instead of looping through all the blocks and doing the same matching over and over for every key like it does now

Reply:
got it